staging/src/k8s.io/client-go/tools/cache/shared_informer_test.go

```diff
@@ -62,16 +87,50 @@ func (l *testListener) OnDelete(obj interface{}) {
 
 func (l *testListener) handle(obj interface{}) {
 	key, _ := MetaNamespaceKeyFunc(obj)
-	fmt.Printf("%s: handle: %v\n", l.name, key)
+	now := l.clock.Now()
+	l.t.Logf("%s %s: handle: %v\n", now.Format(timeFmt), l.name, key)
 	l.lock.Lock()
 	defer l.lock.Unlock()
+	prev, wasThere := l.whenReceived[key]
+	l.whenReceived[key] = now
+	if l.checkTiming && wasThere && (l.resyncPeriod <= 0 || now.Sub(prev) < l.resyncPeriod) {
+		l.t.Errorf("%s %s: prev receipt of %v was at %s but requested period is %s", now.Format(timeFmt), l.name, key, prev.Format(timeFmt), l.resyncPeriod)
+	}
+}
 
-	objectMeta, _ := meta.Accessor(obj)
-	l.receivedItemNames = append(l.receivedItemNames, objectMeta.GetName())
+func (l *testListener) checkRecency(checkPeriod time.Duration) {
+	for key := range l.expectedItemKeys {
+		var prev, now time.Time
+		var good bool
+		// time since last receipt should not go above this quantity
+		expectedResyncPeriod := checkPeriod * time.Duration(math.Ceil(float64(l.resyncPeriod)/float64(checkPeriod)))
+		err := wait.PollImmediate(100*time.Millisecond, 2*time.Second, func() (bool, error) {
```

Comment:
This code *seems* like it's trying to ensure some maximum interval between updates, but it'll return 0 on the iteration that triggers a resync, which seems to kinda defeat the point.  If that's intended, it needs to be explained.